### PR TITLE
CreditNote Entity Update and ContactPersons fixed on Contact Entity

### PR DESCRIPTION
**Pull request by SidneyAllen**
Updated CreditNote Entity to match style of Invoice Entity
Added saveCreditNote method
Added PrivateKey check (try/catch) in oauth.js prior to signing.

Code change below
try {
            var keyCheck = new NodeRSA(key);
           hash = crypto.createSign("RSA-SHA1").update(signatureBase).sign(key, 'base64');
        }
        catch (error) {
            console.log("Your Private Key does not appear to be valid - please check");
        }

### CODE REVIEW

**Comment by SidneyAllen** [comment]
@philals this PR is to address Issue #116 

Thoughts?

**Comment by sandorvasas** [comment]
An explicit checkKeys function would be very cool!
Say I have a use case when the user enters the keys in a form and save it. I'm doing now getAccounts() to check the keys, but it's not the best way. 
Any error in keys leaves the form submission pending. Checking key, secret, and private key explicitly would be the much handier.

**Comment by philals** [comment]
Looks good @SidneyAllen 

Are you able to add a test for catching any exceptions?

**Comment by sandorvasas** [comment]
@Sydney Thanks, I'm aware of the partner program, however entering keys was my client's decision.

**Comment by SidneyAllen** [comment]
@philals - I've removed the RSANode privateKey check from oauth.js file for now.

I want to merge these smaller changes while figuring out how we could test the RSANode code.

File: package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
   "license": "MIT",
   "dependencies": {
     "follow-redirects": "^1.2.3",
-    "lodash": "^4.17.4"
+    "lodash": "^4.17.4",
+    "node-rsa": "^0.4.2"
```
Comment (dupski) [inline package.json]: Hi @SidneyAllen , the `node-rsa` dependency should probably be removed also as it is not used

Comment (SidneyAllen) [reply package.json]: @dupski - removed the node-rsa dependency and merged.

**Comment by philals** [comment]
Cheers @SidneyAllen  I have released this code.

Will write a test node-rsa during next jog day as I think it's a good addition.

**Pull request merged**